pkg/internal/webhookgenerator/server.go

```diff
@@ -0,0 +1,113 @@
+/*
+Copyright 2018 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package webhook
+
+import (
+	"fmt"
+
+	apitypes "k8s.io/apimachinery/pkg/types"
+	types "sigs.k8s.io/controller-runtime/pkg/internal/webhookgenerator/types"
+)
+
+// ServerOptions are options for configuring an admission webhook server.
+type ServerOptions struct {
+	// registry maps a path to a http.Handler.
+	registry map[string]Webhook
+
+	// Port is the port number that the server will serve.
+	// It will be defaulted to 443 if unspecified.
+	Port int32
+
+	// CertDir is the directory that contains the server key and certificate.
+	// If using FSCertWriter in Provisioner, the server itself will provision the certificate and
+	// store it in this directory.
+	// If using SecretCertWriter in Provisioner, the server will provision the certificate in a secret,
+	// the user is responsible to mount the secret to the this location for the server to consume.
+	CertDir string
```

Comment:
I think most of these comments need to be re-written from the new purely generation perspective. Currently they read as if webhook server itself is going to use these options to install webhook configuration.